spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.cloud.firestore.PublicClassMapper;
+import com.google.firestore.v1.DeleteDocumentRequest;
+import com.google.firestore.v1.FirestoreGrpc;
+import com.google.firestore.v1.ListDocumentsRequest;
+import com.google.firestore.v1.ListDocumentsResponse;
+import com.google.protobuf.Empty;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.gcp.data.firestore.util.ObservableReactiveUtil;
+import org.springframework.core.annotation.AnnotationUtils;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class FirestoreTemplate implements FirestoreOperations {
+	private final FirestoreGrpc.FirestoreStub firestore;
+
+	private final String parent;
+
+	public FirestoreTemplate(FirestoreGrpc.FirestoreStub firestore, String parent) {
+		this.firestore = firestore;
+		this.parent = parent;
+	}
+
+
+	public <T> Mono<List<T>> findAll(Class<T> clazz) {
+		Mono<ListDocumentsResponse> listDocumentsResponseMono = runListDocumentsResponse(clazz);
+		return listDocumentsResponseMono
+				.flatMap(listDocumentsResponse -> {
```

Comment:
If the return type becomes Flux<T>, you would do `listDocumentsResponseMono.flatMapMany()`, and then would not have to collect a java stream to a list at all.

Reply:
We get a `List` from upstream, so in reality it is really a `Mono` of a `List`, not a `Flux` of documents